### PR TITLE
Add postgresql 9.6.1

### PR DESCRIPTION
**Pull request by pyhalov**

### CODE REVIEW

File: components/database/postgresql-96/Makefile [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2012-2013, EveryCity Ltd. All rights reserved.
```
Comment (jimklimov) [inline components/database/postgresql-96/Makefile:12]: Add yours )

File: components/database/postgresql-96/Makefile [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+COMPONENT_MJR_VERSION=	9.6
+COMPONENT_MNR_VERSION=	1
+COMPONENT_VERSION=	$(COMPONENT_MJR_VERSION).$(COMPONENT_MNR_VERSION)
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.bz2
+COMPONENT_ARCHIVE_HASH=	sha256:e5101e0a49141fc12a7018c6dad594694d3a3325f5ab71e93e0e51bd94e51fcd
+COMPONENT_ARCHIVE_URL=	http://ftp.postgresql.org/pub/source/v$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
+COMPONENT_PROJECT_URL=  http://www.postgresql.org/
+
+include $(WS_TOP)/make-rules/prep.mk
```
Comment (jimklimov) [inline components/database/postgresql-96/Makefile]: Use proper macro :) `$(WS_MAKE_RULES)`

File: components/database/postgresql-96/Makefile [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+include $(WS_TOP)/make-rules/ips.mk
+
+PATH=	/usr/bin:/usr/sbin:/usr/gnu/bin
+
+# Missing files in build dir for configure without this.
+COMPONENT_PRE_CONFIGURE_ACTION =        ($(CLONEY) $(SOURCE_DIR) $(@D))
+
+CONFIGURE_SCRIPT=       $(@D)/configure
+
+CONFIGURE_OPTIONS = --prefix=/usr/postgres/$(COMPONENT_MJR_VERSION)
```
Comment (jimklimov) [inline components/database/postgresql-96/Makefile:39]: Perhaps build on `shared-macros`'s macros like `PG_HOME=$(USRDIR)/$(PG_IMPLEM)/$(PG_VERSION)` ?

File: components/database/postgresql-96/Makefile [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+CONFIGURE_OPTIONS.32 += --libexecdir=/usr/postgres/$(COMPONENT_MJR_VERSION)/bin/$(MACH32)
+CONFIGURE_OPTIONS.32 += --sbindir=/usr/postgres/$(COMPONENT_MJR_VERSION)/bin/$(MACH32)
+CONFIGURE_OPTIONS.32 += --libdir=/usr/postgres/$(COMPONENT_MJR_VERSION)/lib
+CONFIGURE_OPTIONS.32 += --with-libs=/lib:/usr/lib
+
+CONFIGURE_OPTIONS.64 += --bindir=/usr/postgres/$(COMPONENT_MJR_VERSION)/bin
+CONFIGURE_OPTIONS.64 += --libexecdir=/usr/postgres/$(COMPONENT_MJR_VERSION)/bin
+CONFIGURE_OPTIONS.64 += --sbindir=/usr/postgres/$(COMPONENT_MJR_VERSION)/bin
+CONFIGURE_OPTIONS.64 += --libdir=/usr/postgres/$(COMPONENT_MJR_VERSION)/lib/$(MACH64)
+CONFIGURE_OPTIONS.64 += --with-libs=/lib/$(MACH64):/usr/lib/$(MACH64)
```
Comment (jimklimov) [inline components/database/postgresql-96/Makefile]: Collapse 32 and 64-bit cases with `$(USRLIBDIR.$(BITS))` and similar stuff?

File: components/database/postgresql-96/files/postgresql_96.xml [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+ ident	"@(#)postgresql_96.xml	1.1	09/10/13 SMI"
+
+        NOTE:  This service manifest is not editable; its contents will
+        be overwritten by package or patch operations, including
+        operating system upgrade.  Make customizations in a different
+        file.
+-->
+
+<service_bundle type='manifest' name='postgresql_95'>
```
Comment (jimklimov) [inline components/database/postgresql-96/files/postgresql_96.xml]: 96 now

File: components/database/postgresql-96/files/postgresql_96.xml [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        	</property_group>
+
+        </instance>
+
+        <stability value='Evolving' />
+
+        <template>
+                <common_name>
+                        <loctext xml:lang='C'>
+                                PostgreSQL RDBMS version 9.5
```
Comment (jimklimov) [inline components/database/postgresql-96/files/postgresql_96.xml]: 96

File: components/database/postgresql-96/Makefile [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+COMPONENT_MJR_VERSION=	9.6
+COMPONENT_MNR_VERSION=	1
+COMPONENT_VERSION=	$(COMPONENT_MJR_VERSION).$(COMPONENT_MNR_VERSION)
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.bz2
+COMPONENT_ARCHIVE_HASH=	sha256:e5101e0a49141fc12a7018c6dad594694d3a3325f5ab71e93e0e51bd94e51fcd
+COMPONENT_ARCHIVE_URL=	http://ftp.postgresql.org/pub/source/v$(COMPONENT_VERSION)/$(COMPONENT_ARCHIVE)
+COMPONENT_PROJECT_URL=	http://www.postgresql.org/
+
+include $(WS_TOP)/make-rules/prep.mk
```
Comment (jimklimov) [inline components/database/postgresql-96/Makefile]: WS_MAKE_RULES ?

File: components/database/postgresql-96/Makefile (hunk 3 above)
Comment (pyhalov) [reply components/database/postgresql-96/Makefile:39]: Only macros which I can use here, is $(USRDIR), but /usr is more brief :) 
Another macros you mention refer default values, used paths - current implementation/version.

File: components/database/postgresql-96/Makefile (hunk 4 above)
Comment (pyhalov) [reply components/database/postgresql-96/Makefile]: perhaps, with-libs is not necessary at all... Let's check.

**Comment by jimklimov** [comment]
LGTM now

**Pull request merged**